activiti-services/activiti-services-query/activiti-services-query-rest/src/test/java/org/activiti/services/query/rest/EnableActivitiRestQueryServiceTest.java

```diff
@@ -0,0 +1,168 @@
+		}
+	}
+	
+	@Before
+	public void setUp() {
+		this.mvc = MockMvcBuilders.webAppContextSetup(context).build();
+		this.client = new TestMvcClient(mvc, discoverers).setBasePath(basePath);
+	}
+	
+	@Test
+	public void contextLoads() {
+		// should pass
+	}
+
+	@Test
+	public void exposesAlpsCollectionResourcesWithIdsAndAssociations() throws Exception {
+
+		// Given 
+		Link profileLink = client.discoverUnique("profile");
+		
+		// When
+		Link resourceLink = client.discoverUnique(profileLink, "process-instances", MediaType.ALL);
+
+		// Then
+		client.follow(resourceLink, RestMediaTypes.ALPS_JSON)//
+			.andExpect(jsonPath("$.alps.version").value("1.0"))//
+			.andExpect(jsonPath("$.alps.descriptors[*].name", hasItems("process-instances", "process-instance")))
+			.andExpect(jsonPath("$.alps.descriptors[0].descriptors[*].name", hasItems("processInstanceId", "tasks", "variables")))
+		;
+
+		// When
+		resourceLink = client.discoverUnique(profileLink, "tasks", MediaType.ALL);
+
+		// Then
+		client.follow(resourceLink, RestMediaTypes.ALPS_JSON)//
+			.andExpect(jsonPath("$.alps.version").value("1.0"))//
+			.andExpect(jsonPath("$.alps.descriptors[*].name", hasItems("tasks", "task")))
+			.andExpect(jsonPath("$.alps.descriptors[0].descriptors[*].name", hasItems("id", "processInstance", "variables")))
+		;
+
+		// When
+		resourceLink = client.discoverUnique(profileLink, "variables", MediaType.ALL);
+
+		// Then
+		client.follow(resourceLink, RestMediaTypes.ALPS_JSON)//
+			.andExpect(jsonPath("$.alps.version").value("1.0"))//
+			.andExpect(jsonPath("$.alps.descriptors[*].name", hasItems("variables", "variable")))
+			.andExpect(jsonPath("$.alps.descriptors[0].descriptors[*].name", hasItems("id", "processInstance", "task")))
+		;
+		
+	}
+
+	@Test
+	public void deleteProcessInstancesNotAllowed() throws Exception {
+		deleteResourceIsNotAllowed("process-instances");
+	}	
+
+	@Test
+	public void deleteTasksNotAllowed() throws Exception {
+		deleteResourceIsNotAllowed("tasks");
+	}	
+
+	@Test
+	public void deleteVariablesNotAllowed() throws Exception {
+		deleteResourceIsNotAllowed("variables");
+	}	
+	
+	@Test
+	public void getPagingAndSortingProcessInstancesIsOK() throws Exception {
+		getPagingAndSortingResourceQueryIsOK("process-instances");
+	}	
+
+	@Test
+	public void getPagingAndSortingTasksIsOK() throws Exception {
+		getPagingAndSortingResourceQueryIsOK("tasks");
+	}	
+
+	@Test
+	public void getPagingAndSortingVariablesIsOK() throws Exception {
+		getPagingAndSortingResourceQueryIsOK("variables");
+	}
+	
+	private void getPagingAndSortingResourceQueryIsOK(String collection) throws Exception {
+		mvc.perform(get(basePath+"/"+collection))
+			.andExpect(status().isOk())
+			.andExpect(jsonPath("$._links.self.href", endsWith(collection+"{?page,size,sort}")));
+	}
+	
+	private void deleteResourceIsNotAllowed(String resourceRel) throws Exception {
+		mvc.perform(delete(basePath+"/"+resourceRel+"/0"))
+			.andExpect(status().is(405));

```

Comment:
Small detail: what do you think about using `.isMethodNotAllowed()` instead of `.is(405)`?

Reply:
OK